### PR TITLE
GRW-1337-refact: ProductSelector

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## What?

- rename visual component that renders the insurance choice cards: `ProductSelector` --> `Selector`
- added a new component - `ProductSelector` - that proper parse quotecart data and renders `Selector` 

## Why?

So it becomes easy to integrate it into _offer_ page

**Ticket(s): [GRW-1337](https://hedvig.atlassian.net/browse/GRW-1337)**

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1121** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1121" target="_blank">Graphite</a> 
    * **PR #1122** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1122" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1122?utm_source=stack-comment).

File: src/client/pages/Offer/ProductSelector/Selector.tsx [hunk 1]
```diff
@@ -115,13 +148,19 @@ export const ProductSelector = ({
               image={image}
               checked={selectedProductsByCategory.additionalProducts[id]}
               onClick={() => {
```
Comment (robinandeer) [inline src/client/pages/Offer/ProductSelector/Selector.tsx:150]: nit: this looks like code duplication - maybe consider extracting to a function but could be done later also

**Review by robinandeer — APPROVED** [review]
looks good to me

kind of hard to wrap my head around the code but okey for this code base and right now

[cathack](https://app.graphite.dev/macros/HedvigInsurance)

File: src/client/pages/Offer/ProductSelector/Selector.tsx [hunk 2]
```diff
@@ -1,63 +1,88 @@
...
+
+  const { search: searchParams } = useLocation()
+  const history = useHistory()
+
   const inputRef = useRef<HTMLInputElement | null>(null)
+
   const [selectedProductsByCategory, setSelectedProductsByCategory] = useState(
-    getInitialState({ mainProducts, additionalProducts }),
+    getStateFromURL({ standaloneProducts, additionalProducts, searchParams }),
   )
```
Comment (robinandeer) [inline src/client/pages/Offer/ProductSelector/Selector.tsx:81]: can we get rid of this state and just rely on the URL instead?

would be good with one source of truth

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1122?utm_source=stack-comment).


Job Reference: zxLtjZlK3VtfUNAuwuaI

**Pull request merged**